internal/model/rwfolder.go

```diff
@@ -1402,3 +1432,57 @@ func moveForConflict(name string) error {
 	}
 	return err
 }
+
+func (p *rwFolder) newError(path string, err error) {
+	p.errorsMut.Lock()
+	defer p.errorsMut.Unlock()
+
+	// We might get more than one error report for a file (i.e. error on
+	// Write() followed by Close()); we keep the first error as that is
+	// probably closer to the root cause.
+	for i := range p.errors {
+		if p.errors[i].Path == path {
+			return
+		}
+	}
```

Comment:
A `map[string]error` might fit here better?
You just check if it's present, and if it is, don't overwrite it.